app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,26 @@
+<section><h1>Recipes</h1>
+  <h3><%=@recipe.name %></h3>
+  <p>Complexity: <%= @recipe.complexity %></p>
+  <p>Genre: <%= @recipe.genre %></p>
+</section>
+
+<section>
+  <h4>Ingredients</h4>
+  <p><% @ingredients.each do |ingredient| %></p>
+  <p><%= ingredient.name %></p>
+  <% end %>
+</section>
+
+<div>
+  <p>Total Cost:<%= @ingredients.sum { |i| i.cost } %></p>
```

Comment:
We don't want to be performing calculations in our view. I'd write a method in your `Recipe` class to do this calculation and then just call the method in the view. The other thing is you're using plain Ruby here and we should try to use ActiveRecord when we can. It's going to be much more efficient for us. Something like this:
```ruby
class Recipe
  ...relationships and validations here...

  def total_cost
   ingredients.sum(:cost)
  end
end
```
and then in the view you'd do:
```ruby
  <p>Total Cost:<%= @recipe.total_cost %></p>
```